packages/core/admin/_internal/node/develop.ts

```diff
@@ -230,20 +230,27 @@ const develop = async ({
     loadStrapiSpinner.text = `Loading Strapi (${prettyTime(loadStrapiDuration)})`;
     loadStrapiSpinner.succeed();
 
-    timer.start('generatingTS');
-    const generatingTsSpinner = logger.spinner(`Generating types`).start();
-
-    await tsUtils.generators.generate({
-      strapi: strapiInstance,
-      pwd: cwd,
-      rootDir: undefined,
-      logger: { silent: true, debug: false },
-      artifacts: { contentTypes: true, components: true },
-    });
+    // if this is a JS project (no tsconfig) AND they have disabled type autogeneration, skip type generation
+    if (!tsconfig?.config && strapi.config.get('typescript.autogenerate') === false) {
+      logger.log('Skipping typescript autogeneration');
+    }
```

Comment:
Is this way round correct? i.e. you couldn't turn it off if you're in a TS project right? But you _might_ want too? and you have to set the config to turn it off in a JS project which seems like extra steps?

Also do we need to log this? If so I would use `info` not `log` & if not we can just use `debug`?